ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ip-dhcp-snooping.j2

```diff
@@ -0,0 +1,27 @@
+{#
+    Copyright (c) 2023 Arista Networks, Inc.
+    Use of this source code is governed by the Apache License 2.0
+    that can be found in the LICENSE file.
+#}
+{# eos - ip dhcp snooping #}
+{% if ip_dhcp_snooping is arista.avd.defined %}
+{%     if ip_dhcp_snooping.enabled is arista.avd.defined(true) %}
+{%         if ip_dhcp_snooping.bridging is arista.avd.defined(true) %}
+!
+ip dhcp snooping bridging
+{%         else %}
+!
+ip dhcp snooping
+{%         endif %}
+{%         if ip_dhcp_snooping.information_option.enabled is arista.avd.defined(true) %}
+{%             if ip_dhcp_snooping.information_option.circuit_id_type is arista.avd.defined and ip_dhcp_snooping.information_option.circuit_id_format is arista.avd.defined %}
+ip dhcp snooping information option circuit-id type {{ ip_dhcp_snooping.information_option.circuit_id_type }} format {{ ip_dhcp_snooping.information_option.circuit_id_format }}
+{%             else %}
+ip dhcp snooping information option
+{%             endif %}
+{%         endif %}
```

Comment:
Are we sure for this logic block ?

It seems that when I have
```
host3(config)#show run | grep snoop
ip dhcp snooping bridging
ip dhcp snooping information option circuit-id type 24 format %p:%v
```

only, the option is not active:


```
host3(config)#show ip dhcp snooping
DHCP Snooping is enabled
DHCP Snooping is not operational
DHCP Snooping is configured on following VLANs:
  None
DHCP Snooping bridging is operational on following VLANs:
  None
Insertion of Option-82 is disabled
```


But when I configure:
```
host3(config)#show run | grep snoop
ip dhcp snooping bridging
ip dhcp snooping information option
ip dhcp snooping information option circuit-id type 24 format %p:%v
```


then the option 82 becomes active:
```
host3(config)#show ip dhcp snooping
DHCP Snooping is enabled
DHCP Snooping is operational
DHCP Snooping is configured on following VLANs:
  None
DHCP Snooping bridging is operational on following VLANs:
  None
Insertion of Option-82 is enabled
  Circuit-id default sub-option Type: 24
  Circuit-id default format: Interface name:VLAN ID
  Remote-id: 00:1c:73:64:7b:1f (Switch MAC)
```

Reply:
@gmuloc you are absolutely correct. I justed checked it on a switch right now too.
I am sorry, I missed that in my tests.

```
{#
    Copyright (c) 2023 Arista Networks, Inc.
    Use of this source code is governed by the Apache License 2.0
    that can be found in the LICENSE file.
#}
{# eos - ip dhcp snooping #}
{% if ip_dhcp_snooping is arista.avd.defined %}
{%     if ip_dhcp_snooping.enabled is arista.avd.defined(true) %}
{%         if ip_dhcp_snooping.bridging is arista.avd.defined(true) %}
!
ip dhcp snooping bridging
{%         else %}
!
ip dhcp snooping
{%         endif %}
{%         if ip_dhcp_snooping.information_option.enabled is arista.avd.defined(true) %}
ip dhcp snooping information option
{%             if ip_dhcp_snooping.information_option.circuit_id_type is arista.avd.defined and ip_dhcp_snooping.information_option.circuit_id_format is arista.avd.defined %}
ip dhcp snooping information option circuit-id type {{ ip_dhcp_snooping.information_option.circuit_id_type }} format {{ ip_dhcp_snooping.information_option.circuit_id_format }}
{%             endif %}
{%         endif %}
```

Reply:
I have addressed this in an updated template. Please take a look again.

Reply:
@ClausHolbechArista 
I have tested it with your changes again on a 722XPM EOS 4.30.0F successfully:

```
ip_dhcp_snooping:
  enabled: true
  bridging: true
  information_option:
    enabled: true
    circuit_id_type: 10
    circuit_id_format: '%h:%p'
  vlan: "10,20,500,1000-2000"
```

```
access-1h1-01__11:23:14(config)#ip dhcp snooping bridging
access-1h1-01__11:23:43(config)#ip dhcp snooping information option
access-1h1-01__11:23:43(config)#ip dhcp snooping information option circuit-id type 10 format %h:%p
access-1h1-01__11:23:43(config)#ip dhcp snooping vlan 10,20,500,1000-2000
access-1h1-01__11:23:44(config)#show ip dhcp snooping
DHCP Snooping is enabled
DHCP Snooping is operational
DHCP Snooping is configured on following VLANs:
  10,20,500,1000-2000
DHCP Snooping bridging is operational on following VLANs:
  10,20,500,1000-2000
Insertion of Option-82 is enabled
  Circuit-id default sub-option Type: 10
  Circuit-id default format: Hostname:Interface name
  Remote-id: ac:3d:94:c7:d6:ba (Switch MAC)
```

Reply:
Thanks for checking @kmueller68